基于MindSpore NLP实现protein folding应用案例开发

autism-ip opened the pull request:
### PR 描述正文范文

#### 1. 关联 Issue

Fixes #1801

#### 2. 修改描述 

本PR基于 [[Hugging Face Protein Folding 教程](https://github.com/huggingface/notebooks/blob/main/examples/protein_folding.ipynb)](https://github.com/huggingface/notebooks/blob/main/examples/protein_folding.ipynb) **MindSpore 2.7.0** 和 **MindSpore NLP 0.5.1** 框架，完成了蛋白质折叠（Protein Folding）领域的应用案例开发。

主要工作内容包括：

* **模型实现**：基于 MindNLP 调用 EsmForProteinFolding模型并加载hf上对应权重。
* **全流程覆盖**：实现了从氨基酸序列数据预处理、模型加载、模型微调到三维结构预测推理的完整流程。
* **可视化**：集成了蛋白质三维结构的可视化展示（使用 py3Dmol） 。
* **文档规范**：
* 提交了 `.ipynb` 格式的案例文件，内部文本描述均为中文。
* 代码结构符合 `mindspore-courses/applications` 合入规范。
* 同步更新了对应目录下的 README 文档。



#### 3. 开发环境 (Environment)

本次开发与测试在 **Atlas 800I A2 / Atlas 800T A2** 推理服务器上完成。

* **硬件环境**: Ascend NPU
* **操作系统**:  Ubuntu Linux
* **Python 版本**: 3.11.10
* **CANN 版本**: 8.2.RC1 
* **MindSpore 版本**: 2.7.0
* **MindSpore NLP 版本**: 0.5.1

#### 4. 目录结构变更 

```text
applications/
└── EsmForProteinFolding/  
      ├── protein_folding.ipynb      # 核心案例代码
      └── README.md                  

```

#### 5. 效果展示 

* **推理结果**：
成功预测了目标序列的三维结构，生成的 PDB 文件符合预期。
* **功能测试**：
* [x]数据加载正常
* [x] 模型微调流程跑通
* [x] 推理与可视化跑通



#### 6. 自检清单 

* [x] 代码文件为 `.ipynb` 格式，且文本描述为中文。
* [x] 已经过本地运行测试。
* [x] 遵循了 [[MindSpore Applications 合入规范](https://github.com/mindspore-courses/applications/wiki/Contributing-Guidelines)](https://github.com/mindspore-courses/applications/wiki/Contributing-Guidelines)。
* [x] 相关依赖库（如 `biopython`, `py3Dmol` 等）已在 Notebook 安装步骤中注明。

autism-ip committed:
On branch feature/protein-folding
  Changes to be committed:
	new file:   READEME.md
	new file:   protein_folding.ipynb

autism-ip committed:
rename file

autism-ip committed:
delete repeated file

autism-ip commented:
重复文件已删除

autism-ip committed:
On branch feature/protein-folding
 Changes to be committed:
	modified:   EsmForProteinFolding/inference_EsmForProteinFolding_prediction.ipynb

autism-ip committed:
On branch feature/protein-folding
 Changes to be committed:
	deleted:    EsmForProteinFolding/READEME.md
	deleted:    EsmForProteinFolding/inference_EsmForProteinFolding_prediction.ipynb

autism-ip committed:
Merge remote-tracking branch 'upstream/dev' into feature/protein-folding

autism-ip committed:
On branch feature/protein-folding
 Changes to be committed:
	new file:   esmforproteinfolding/inference_esmforproteinfolding_prediction.ipynb

autism-ip committed:
On branch feature/protein-folding
 Changes to be committed:
	modified:   README.md

autism-ip committed:
On branch feature/protein-folding
 Changes to be committed:
	modified:   inference_esmforproteinfolding_prediction.ipynb

xing-yiren commented on llm/esmforproteinfolding/inference_esmforproteinfolding_prediction.ipynb:
```diff
@@ -0,0 +1,493 @@
+    "\n",
+    "# 设置运行环境（默认 Ascend 环境）\n",
+    "# 推荐使用 PYNATIVE_MODE 便于动态形状；如需极致性能可切换 GRAPH_MODE+AMP\n",
+    "ms.set_context(mode=ms.PYNATIVE_MODE, device_target=\"Ascend\")\n",
```
默认情况下走的就是动态图，所以理论上这里不用显式地指定了

xing-yiren commented:
其他的没有了，辛苦把最后一点小细节修改下，赞

autism-ip committed:
On branch feature/protein-folding
 Changes to be committed:
	modified:   inference_esmforproteinfolding_prediction.ipynb

xing-yiren commented:
okie我这边没问题了，辛苦 @wang-hua-2019 也把把关

xing-yiren committed:
Merge branch 'dev' into feature/protein-folding
